### ISSUE TITLE
Antox stays offline after an interruption of the mobile-data (>5min) of the mobil phone (new bug since build 0.23.160!?)

### ISSUE THREAD

**Issue by toxtox** [subliun/Antox#217]
When I used build **0.23.98** I never logged out before I switched off my mobile-data in order to save battery.
And there was no need to take care for mobile-data interruptions caused by weak or no signals (countryside, underground parking, cellar, tube and so on), because 2 or at max. 3 minutes after the mobile-data interruption antox was online again, too.

Now (build **0.23.160** and **0.23.180**) in most cases **antox stays offline** after an mobile-data interruption.

I made an investigation for this issue using build **0.23.180** :

*  **really short mobile-data interruptions** of just 1 minute are **no problem** (switched mobile data off, waited until the *green online dots* of my contact names disappeared, then switched mobile data on and antox was online again 2 minutes later

* **mobile-data interruptions** of more than **5 minutes** or even less cause an **unreliable** behaviour

My test parameters (cycle times) for build **0.23.180** are the following: switching off 20 minutes -> switching on 10 minutes ->  switching off 20 minutes -> and so on
  * Test result 1: antox online again 3 minutes after mobile-data interruptions
  * Test result 2: antox stayed offline
  * Test result 3: antox stayed offline
  * Test result 4: antox online again 9 minutes after mobile-data interruptions
  * Test result 5: antox online again 3 minutes after mobile-data interruptions
  * Test result 6: antox stayed offline
  * Test result 7: antox stayed offline
  * Test result 8: antox stayed offline
  * Test result 9: antox online again 8 minutes after mobile-data interruptions

Maybe antox would have been online again if I had waited longer than 10 minutes but waiting longer than 10 minutes is not acceptable (in my opinion ;-).

The same test procedure I have run with the old build **0.23.98** and it was completely successful (antox was always online again 2 or 3 minutes after mobile-data interruption)

Maybe the fix for the bug #215 caused this unreliable online/offline-behaviour as a side-effect?

**Comment by subliun** [comment]
That's interesting, I'll look into it. Does the same thing happen on wifi or is it exclusive to mobile-data?

**Comment by toxtox** [comment]
Yes it happens with wifi and mobile-data. And I played around with the options " *only WIFI* " and " *activate UDP* " without success.
My test conditions were " *only WIFI* " *off* and *activate UDP* " *on*

**Comment by toxtox** [comment]
I made some further investigations on this issue and the results are the following:

- the old 0.23.98 build contains the bug, too. ( I tried to reproduce the prior successful test again but it failed [The unreliable behaviour depends on your luck ;-) ])
=> I changed the *title* 
=> The changes to fix the #215 issue are obviously NOT responsible for the "offline->online-bug")

- After the mobile device is online again the *green dot* of my own account avatar is always on again, too. BUT the *dots* of the *contact names* stay grey.

- It happens with a peer to peer *qTox (nightly 01.Dec.2015)* <-> *Antox* connection and a pure *Antox* <-> *Antox* connection (**both build 0.23.213**)

- If three TOX - Devices are connected ( 2x Antox and 1x qTox) and one of the Antox-devices goes offline and then online again it happens that both or only one connection to the two other devices or no connection is restored. (The failing connection can be from Antox to Antox als well as from Antox to qTox)

- disabling all the green setup options did not help

- After the logout and the login of one of the devices the *dots* of the *contact names* turn to green 2 or 3 minutes later.

=> **Would it be possible to implement an *auto-logout-login* feature or an *auto-refresh* feature after a mobile-data interruption occurred?
